### PR TITLE
fix: Amphtml error 

### PR DESCRIPTION
**Pull request by shraddha-kesari**
# Description

Ref ticket: https://github.com/quintype/ace-planning/issues/656

Fixes # (issue-reference)

Change amphtml url structure depending on the story template.
If its a visual story, amphtml href will be `domain/story-slug` orelse  it will be `domain/amp/story/story-slug`. 
Because visual stories already have `ampstories/` in their storyslug and thats causing the issue. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

1. Open PrabhatKhabhar [visual story url](https://www.prabhatkhabar.com/entertainment/bhojpuri-actor-khesari-lal-yadav-wife-chanda-beautiful-pics-family-know-about-actor-personal-life-dvy)
2. Open view page source 
3. In the page source, Check for amphtml  url -- it should be https://www.prabhatkhabar.com/ampstories/entertainment/bhojpuri-actor-khesari-lal-yadav-wife-chanda-beautiful-pics-family-know-about-actor-personal-life-dvy

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ags1773** [comment]
add tests pls

File: test/amp_tags_test.js [hunk 1]
```diff
@@ -13,18 +13,30 @@ describe('AmpTags', function () {
...
     const string = getSeoMetadata(seoConfig, config, "story-page", { data: { story: story } }, {});
     assertContains('<link rel="amphtml" href="/amp/story/section%2Fslug"/>', string);
   });
 
+  it("does not rely on is-amp-supported in visual story API", function () {
+    const story = { slug: "section/slug", "is-amp-supported": false };
+    const string = getSeoMetadata(seoConfig, config, "story-page", { data: { story: { ...story, "story-template": "visual-story" } } }, {});
+    assertContains('<link rel="amphtml" href="/section%2Fslug"/>', string);
+  });
+
```
Comment (ags1773) [inline test/amp_tags_test.js:39]: This test isn't needed. `is-amp-supported` is independent of story template

File: test/amp_tags_test.js [hunk 2]
```diff
@@ -56,12 +68,38 @@ describe('AmpTags', function () {
...
     const string = getSeoMetadata(seoConfig, config, "story-page", { data: { story: story } }, {});
     assertContains('<link rel="amphtml" href="/amp/story/section%2Fslug"/>', string);
   });
 
+  it("does not append domain to amp visual stories if appendHostToAmpUrl not present", function () {
+    const story = { slug: "section/slug", "is-amp-supported": true };
+    const string = getSeoMetadata(seoConfig, config, "story-page", { data: { story: { ...story, "story-template": "visual-story" } } }, {});
+    assertContains('<link rel="amphtml" href="/section%2Fslug"/>', string);
+  });
+
```
Comment (ags1773) [inline test/amp_tags_test.js:102]: independent of story template, test is not needed

File: test/amp_tags_test.js [hunk 3]
```diff
@@ -74,6 +112,18 @@ describe('AmpTags', function () {
...
+    const string = getSeoMetadata(
+      { ...seoConfig, appendHostToAmpUrl: true },
+      config,
+      "story-page",
+      { currentHostUrl: "https://madrid.quintype.io/section/slug", data: { story: { ...story, "story-template": "visual-story" } } },
+      {}
+    );
+    assertContains('<link rel="amphtml" href="https://madrid.quintype.io/section%2Fslug"/>', string);
+  });
+
```
Comment (ags1773) [inline test/amp_tags_test.js:126]: independent of story template, test is not needed

File: test/amp_tags_test.js [hunk 4]
```diff
@@ -85,4 +135,16 @@ describe('AmpTags', function () {
...
+    const story = { slug: "section%2Fslug", "is-amp-supported": true };
+    const string = getSeoMetadata(
+      { ...seoConfig, appendHostToAmpUrl: true, decodeAmpUrl: true },
+      config,
+      "story-page",
+      { currentHostUrl: "https://madrid.quintype.io/section%2Fslug", data: { story: { ...story, "story-template": "visual-story" } } },
+      {}
+    );
+    assertContains('<link rel="amphtml" href="https://madrid.quintype.io/section/slug"/>', string);
+  });
```
Comment (ags1773) [inline test/amp_tags_test.js:149]: independent of story template, test is not needed

File: test/amp_tags_test.js [hunk 5]
```diff
@@ -56,12 +68,38 @@ describe('AmpTags', function () {
...
+    const privateStoryResults = getSeoMetadata(
+      { ...seoConfig, ampStoryPages: "public" },
+      config,
+      "story-page",
+      { data: { story: { ...story, access: "subscription", "story-template": "visual-story" } } },
+      {}
+    );
+    assert.equal("", privateStoryResults);
+  });
+
```
[the remark this answers is not in the document]
Comment (shraddha-kesari) [reply test/amp_tags_test.js:90]: just visual stories

File: test/amp_tags_test.js (hunk 1 above)
Comment (shraddha-kesari) [reply test/amp_tags_test.js:39]: Removed.

File: test/amp_tags_test.js [hunk 6]
```diff
@@ -13,18 +13,30 @@ describe('AmpTags', function () {
...
   };
 
-  it("gets amp tags for supported stories", function () {
+  it("gets amp tags for supported stories other than visual stories", function () {
     const story = { slug: "section/slug", "is-amp-supported": true };
     const string = getSeoMetadata(seoConfig, config, "story-page", { data: { story: story } }, {});
     assertContains('<link rel="amphtml" href="/amp/story/section%2Fslug"/>', string);
   });
 
+  it("gets amp tags only for supported visual stories", function () {
```
[the remark this answers is not in the document]
Comment (shraddha-kesari) [reply test/amp_tags_test.js:22]: Done

File: test/amp_tags_test.js [hunk 7]
```diff
@@ -13,18 +13,30 @@ describe('AmpTags', function () {
     "sketches-host": "https://madrid.quintype.io",
   };
 
-  it("gets amp tags for supported stories", function () {
+  it("gets amp tags for supported stories other than visual stories", function () {
```
[the remark this answers is not in the document]
Comment (shraddha-kesari) [reply test/amp_tags_test.js:16]: Okay

File: test/amp_tags_test.js (hunk 2 above)
Comment (shraddha-kesari) [reply test/amp_tags_test.js:102]: removed

File: test/amp_tags_test.js (hunk 3 above)
Comment (shraddha-kesari) [reply test/amp_tags_test.js:126]: removed

File: test/amp_tags_test.js (hunk 4 above)
Comment (shraddha-kesari) [reply test/amp_tags_test.js:149]: removed

**Review by ags1773 — APPROVED** [review]

**Pull request merged**